modules/play-json/src/main/scala/PlayJsonModule.scala

```diff
@@ -0,0 +1,170 @@
+
+      def apply[A](fSchema: FSchema[R.Prim, R.SumTermId, R.ProductTermId, A]): LabelledSchema[A] =
+        (false, fSchema)
+    }
+
+  implicit final def reads(
+    implicit primNT: R.Prim ~> Reads,
+    branchLabel: R.SumTermId <~< String,
+    fieldLabel: R.ProductTermId <~< String
+  ): RInterpreter[Reads] =
+    new Interpreter[R.Prim, R.SumTermId, R.ProductTermId, Reads] {
+
+      val alg = new (HEnvT[Boolean, RSchema, Reads, ?] ~> Reads) {
+
+        def apply[A](schema: HEnvT[Boolean, RSchema, Reads, A]): Reads[A] = schema.fa match {
+          case One() =>
+            Reads {
+              case JsNull => JsSuccess(())
+              case _      => JsError(Seq(JsPath -> Seq(JsonValidationError("error.expected.null"))))
+            }
+          case :+:(left, right) =>
+            Reads(
+              json =>
+                left
+                  .reads(json)
+                  .fold(
+                    el =>
+                      right.reads(json).map(\/-.apply) match {
+                        case JsError(er) => JsError(JsError.merge(el, er))
+                        case x           => x
+                      },
+                    a => JsSuccess(-\/(a))
+                  )
+            )
+          case p: :*:[Reads, a, b, R.Prim, R.SumTermId, R.ProductTermId] =>
+            if (schema.ask)
+              p.left.and(p.right)((x: a, y: b) => (x, y))
+            else
+              (JsPath \ "_1")
+                .read(p.left)
+                .and((JsPath \ "_2").read(p.right))((x: a, y: b) => (x, y))
+          case PrimSchema(p)       => primNT(p)
+          case SumTerm(id, schema) => undefinedAsNull(branchLabel(id), schema)
+          case u: Union[R.Prim, R.SumTermId, R.ProductTermId, Reads, A, a] =>
+            u.choices.map(u.iso.get)
+          case ProductTerm(id, schema) => undefinedAsNull(fieldLabel(id), schema)
+          case r: Record[R.Prim, R.SumTermId, R.ProductTermId, Reads, A, a] =>
+            r.fields.map(r.iso.get)
+          case SeqSchema(elem) =>
+            Reads {
+              case JsArray(elems) =>
+                elems.toList.traverse(elem.reads _)
+              case _ => JsError(Seq(JsPath -> Seq(JsonValidationError("error.expected.jsarray"))))
+            }
+          case i: IsoSchema[R.Prim, R.SumTermId, R.ProductTermId, Reads, a0, A] =>
+            i.base.map(i.iso.get)
+        }
+      }
+
+      def interpret = hyloNT(labelRecordFields, alg).compose(labellingSeed)
+    }
+
+  implicit final def writes(
+    implicit primNT: R.Prim ~> Writes,
+    branchLabel: R.SumTermId <~< String,
+    fieldLabel: R.ProductTermId <~< String
+  ): RInterpreter[Writes] =
+    new Interpreter[R.Prim, R.SumTermId, R.ProductTermId, Writes] {
+
+      val alg = new (HEnvT[Boolean, RSchema, Writes, ?] ~> Writes) {
+
+        def apply[A](env: HEnvT[Boolean, RSchema, Writes, A]): Writes[A] = env.fa match {
+          case One()            => Writes(_ => JsNull)
```

Comment:
Do you think it would also be possible to not emit optional fields that have the `None` value?

Reply:
I'm not completely sure it would be impossible, but I'm unable to devise a way to do so (for now).

That's because of the encoding of options as a mere `Iso` on top of `A \/ Unit`. Not emitting fields valued as `None` would therefore require to find a way to process `x :+: y` differently wether `y` is `One()` or something else.

So I guess your remark makes another case for adding an `Optional` member to the `Schema` GADT.

Reply:
> So I guess your remark makes another case for adding an `Optional` member to the `Schema` GADT.

That’s what I also think.

Reply:
This is what I was saying in my PR regarding `A \/ Unit` can lead to an ugly representation. The Optional member for the GADT should be trivial but we should take care we don't pollute the GADT too much with such special cases. This brings us full circle back to the discussion in gitter about effects in schemas :grin: